<launch>

  <!--node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" /-->
  
  <!--node name="keyboard_control" pkg="keyboard_control" type="keyboard_control_node" output="screen">
    <param name="linear_init" value="0.0" />
    <param name="linear_max" value="1" />
    <param name="linear_step" value="0.05" />

    <param name="angular_init" value="1" />
    <param name="angular_max" value="1" />
    <param name="angular_step" value="0.1" />
  </node-->

  <node name="zhongqi_driver" pkg="zhongqi_driver" type="zhongqi_driver_node" output="screen">
    <param name="port_name" value="/dev/ucar" />
    <param name="control_rate" value="10" />
    <param name="check_pos_rate" value="1" />
    <param name="baud_rate" value="115200" />
    <param name="wheelbase" value="1.07" />
    <param name="tread" value="0.685" />
    <param name="wheel_diameter" value="0.37" />
    <param name="maxsteer" value="0" />
    <param name="steer_radius" value="true" />
    <param name="publish_tf" value="true" />
    <param name="check" value="false" />
    <param name="odom_frame" value="odom"/>
    <param name="base_frame" value="base_footprint"/>
    <param name="vth_c" value="1.0" />
    <param name="vx_c" value="1.146" />
    <param name="vy_c" value="1" />
    <param name="vth_c" value="1" />
    <param name="serial_db" value="0" /><!--0:n 1:r 2/4:s 3:a -->
  </node>

  <node pkg="marvel" type="marvel_driver" name="marvelmind_driver" output="screen" respawn="true">
    <remap from="/marvelmind_hedge_pos" to="location_pos"/>
    <param name="port_name" value="/dev/urear"/>
    <param name="baud_rate" value="115200"/>
    <param name="address" value="16"/>
	<param name="map_frame" value="tag"/>
  </node>

  <node pkg="marvel" type="marvel_driver" name="marvelmind_driver_2" output="screen" respawn="true">
    <remap from="/marvelmind_hedge_pos" to="location_pos_2"/>
    <param name="port_name" value="/dev/ufront"/>
    <param name="baud_rate" value="115200"/>
    <param name="address" value="14"/>
    <param name="map_frame" value="tag2"/>
  </node>

  <node pkg="ah100b" type="ah100b" name="imu" respawn="false">
    <param name="port" value="/dev/uimu" type="string" />
    <param name="bund" value="115200" type="int" />
    <remap from="/imu" to="imu"/>
  </node>

  <!--node pkg="ah100b" type="ah100b" name="imu_2" respawn="false">
    <param name="port" value="/dev/uimu-2" type="string" />
    <param name="bund" value="115200" type="int" />
    <remap from="/imu" to="imu_2"/>
  </node-->

  <node pkg="wr_fslidar"  type="fslidar_node" name="fslidar_front" output="screen">
  	<param name="dev_path"         type="string" value="/dev/ulidar_front"/>
  	<param name="inverted"            type="bool"   value="false"/>
  	<param name="frame_id"            type="string" value="front_laser_link"/>
   	<param name="sample_rate"         type="int"    value="5000"/>
   	<param name="rotational_speed"    type="int"    value="5"/>
   	<param name="scan_topic"            type="string"   value="/scan_front"/>
  </node>

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="lidar_filter_front">
    <remap from="/scan" to="/scan_front" />
    <remap from="/scan_filtered" to="/scan_1" />
    <rosparam command="load" file="$(find wr_fslidar)/param/front_lidar_config.yaml" />
  </node>

  <node pkg="wr_fslidar"  type="fslidar_node" name="fslidar_back" output="screen">
  	<param name="dev_path"         type="string" value="/dev/ulidar_rear"/>
   	<param name="inverted"            type="bool"   value="false"/>
   	<param name="frame_id"            type="string" value="rear_laser_link"/>
   	<param name="sample_rate"         type="int"    value="5000"/>
   	<param name="rotational_speed"    type="int"    value="5"/>
   	<param name="scan_topic"            type="string"   value="/scan_back"/>
  </node>

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="lidar_filter_back">
    <remap from="/scan" to="/scan_back" />
    <remap from="/scan_filtered" to="/scan_2" />
    <rosparam command="load" file="$(find wr_fslidar)/param/back_lidar_config.yaml" />
  </node>
 
  <group ns="/usb_cam">
    <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen" respawn="false">
      <param name="width" type="int" value="320" />
      <param name="height" type="int" value="240" />
      <param name="fps" type="int" value="15" />
      <param name="frame" type="string" value="wide_stereo" />

      <param name="auto_focus" type="bool" value="False" />
      <param name="focus_absolute" type="int" value="0" />

      <param name="device" type="string" value="/dev/ucamera_front" />
      <param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example.yaml" />
    </node>
  </group>
  <group ns="/usb_cam2">
    <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera_2" output="screen" respawn="false">
      <param name="width" type="int" value="320" />
      <param name="height" type="int" value="240" />
      <param name="fps" type="int" value="15" />
      <param name="frame" type="string" value="wide_stereo" />

      <param name="auto_focus" type="bool" value="False" />
      <param name="focus_absolute" type="int" value="0" />

      <param name="device" type="string" value="/dev/ucamera_rear" />
      <param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example.yaml" />
      <remap from="/usb_cam2/image_raw/" to="/image_raw1/"/>

    </node>
  </group>
   <!--include file="$(find video_stream_opencv)/launch/camera.launch">
    <arg name="video_stream_provider" value="2" />
  	<arg name="set_camera_fps" default="30" />
  </include-->
</launch>
